xmppserver/src/main/java/org/jivesoftware/openfire/http/HttpSession.java

```diff
@@ -979,22 +991,22 @@ private void closeSession(@Nullable final StreamError error) {
             // open connections in this method.
             synchronized (connectionQueue) {
                 boolean isFirst = true;
-                for (HttpConnection toClose : connectionQueue) {
+                while (!connectionQueue.isEmpty()) {
+                    final HttpConnection toClose = connectionQueue.poll();
+                    assert !toClose.isClosed();
```

Comment:
assert?

Reply:
Same rationale as above. The statement asserts the same condition: the queued connections should not be in a 'closed' state.